.github/workflows/docker.yaml

```diff
@@ -13,45 +15,145 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  build:
-    name: Build and push dev image
+  tag:
+    name: Create image tag
     runs-on: ubuntu-latest
+    outputs:
+      subject: ${{ steps.image_tag.outputs.subject }}
+      tag: ${{ steps.image_tag.outputs.tag }}
+    steps:
+      - name: Use ghcr.io
+        if: github.event_name != 'push' || github.ref_type != 'tag'
```

Comment:
It is a bit difficult for me to parse these if clauses on every step, when they are kind of if-else, is it possible to have two jobs, one for pull requests which use GHCR and another for Push+Tag for Docker.io, or is it more tricky than that?

Reply:
I don't think we should change this part. The `tag` job was made to make the rest of the workflow less if/else-y. If we were to duplicate it, we'd have to duplicate the entire build & release process as well, which seems _very_ error-prone.